.github/workflows/ci.yaml

```diff
@@ -109,98 +105,133 @@ jobs:
           node-version: 20
       - uses: extractions/setup-just@v1
       - uses: actions/cache@v3
-        name: Cache
+        name: cache
         with:
           path: |
             ~/.cargo/
             target/
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-      - name: Build nodejs
+      - name: build node.js bindings
         run: just js build-debug
-      - name: Node Bindings Tests
+      - name: run node.js binding tests
         run: just js test
+
   pg-protocol:
     name: PG Protocol Tests
     runs-on: ubuntu-latest-8-cores
     needs: ["build"]
     steps:
-      - name: Checkout
+      - name: checkout
         uses: actions/checkout@v4
       - uses: extractions/setup-just@v1
       - uses: actions/cache@v3
-        name: Cache
+        name: cache
         with:
           path: |
             ~/.cargo/
             target/
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-      - name: PG Protocol Tests
+      - name: pg protocol test (script)
         run: |
           PROTOC=`just protoc && just --evaluate PROTOC` ./scripts/protocol-test.sh
 
-      - name: PG Protocol Tests (SLT Runner)
+      - name: pg protocol tests (slt runner)
         run: just slt -v 'pgproto/*'
 
   sql-logic-tests:
     name: SQL Logic Tests
     runs-on: ubuntu-latest-8-cores
     needs: ["build"]
     steps:
-      - name: Checkout
+      - name: checkout
         uses: actions/checkout@v4
       - uses: extractions/setup-just@v1
       - uses: actions/cache@v3
-        name: Cache
+        name: cache
         with:
           path: |
             ~/.cargo/
             target/
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
 
-      - name: Public SQL Logic Tests DEBUG
+      - name: public sql logic tests DEBUG
         if: ${{ env.ACTIONS_STEP_DEBUG == 'true' }}
         run: just slt -v 'sqllogictests/*'
-      - name: Public SQL Logic Tests
+      - name: public sql logic tests
         if: ${{ env.ACTIONS_STEP_DEBUG != 'true' }}
         run: just slt 'sqllogictests/*'
 
-  integration-tests:
+  process-integration-tests:
+    name: Process Integration Tests (pytest)
+    runs-on: ubuntu-latest-8-cores
+    needs: ["build"]
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+      - uses: extractions/setup-just@v1
+      - name: install poetry
+        uses: snok/install-poetry@v1
+      - name: install python
+        uses: actions/setup-python@v5
+        with:
+            python-version: '3.11' 
+            cache: poetry
+            cache-dependency-paths: |
+              tests/poetry.lock
```

Comment:
musing: something I've found recently with our npm bindings tests, is that the dependency overhead is so small that there's no benefit to caching, and in fact it created a bug when run infrequently and upgrading to modern yarn. I don't think that's a concern here, and I don't know the dependency overhead of these tests. In short, I'd consider not adding this until needed, unless it already is!

Reply:
recompiling the c-parts of the drivers is enough (and change rarely enough) that I wouldn't be super worried about cache bugs or lack of utility here. 

I also think it does caching regardless, but we miss it by having things scoped to the directory rather than the repo.